f5_openstack_agent/lbaasv2/drivers/bigip/test/test_vlan.py

```diff
@@ -0,0 +1,143 @@
+# coding=utf-8
+# Copyright 2014-2016 F5 Networks Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from f5_openstack_agent.lbaasv2.drivers.bigip.network_helper import \
+    NetworkHelper
+
+from f5.bigip.tm.net.vlan import TagModeDisallowedForTMOSVersion
+
+import mock
+import pytest
+
+
+tagged_vlan_no_int = {'name': "test_valan",
+                      'partition': "Project_123456789",
+                      'tag': 1000}
+
+
+class TestVLANCreate(object):
+
+    @pytest.fixture
+    def bigip(self):
+        bigip = mock.MagicMock()
+        bigip.tm.net.vlans.vlan = mock.MagicMock()
+        bigip.tm.net.vlans.vlan.exists.return_value = True
+
+        return bigip
+
+    @pytest.fixture
+    def network_helper(self):
+        nh = NetworkHelper()
+        nh.get_route_domain_by_id = mock.MagicMock(return_value=0)
+
+        return nh
+
+    def test_create_vlan_no_name(self, bigip, network_helper):
+        tagged_vlan_no_name = {'partition': "Project_123456789"}
+
+        vlan = network_helper.create_vlan(bigip, tagged_vlan_no_name)
+
+        assert(vlan is None)
+
+    def test_create_vlan_exists(self, bigip, network_helper):
+        tagged_vlan_no_int = {'name': "test_vlan",
+                              'partition': "Project_123456789",
+                              'tag': 1000}
+
+        v = network_helper.create_vlan(bigip, tagged_vlan_no_int)
+
+        assert(v is not None)
+
+        bigip.tm.net.vlans.vlan.exists.assert_called_once_with(
+            name='test_vlan', partition='Project_123456789')
+        bigip.tm.net.vlans.vlan.load.assert_called_once_with(
+            name='test_vlan', partition='Project_123456789')
+
+    def test_create_vlan_no_int(self, bigip, network_helper):
+
+        tagged_vlan_no_int = {'name': "test_vlan",
+                              'partition': "Project_123456789",
+                              'tag': 1000}
```

Comment:
Ah no interface. I think this test could be clearer with some additional assertions regarding how no interface is created.